Allow overriding batch size for bulk API (#2813)
SFDO-Tooling/CumulusCI: docs/data.rst

specify the key ``bulk_mode: Serial`` in each step requiring the use of serial mode.

> For REST API and smart-API modes, you can specify a batch size using the ``batch_size`` key.

Reviewer: ```suggestion
For all API modes, you can specify a batch size using the ``batch_size`` key.
```

> Allowed values are between 1 and 200 for the REST API and 1 and 10000 for the Bulk API, 

Reviewer: ```suggestion
Allowed values are between 1 and 200 for the REST API and 1 and 10,000 for the Bulk API. Note that the semantics of batch sizes differ between the REST API and the Bulk API. In the REST API, the batch size is the actual size of individual transactions. In the Bulk API, the batch size is the maximum record count in a Bulk API batch, which is subject to its own limits, including restrictions on total processing time. Bulk API batches are automatically chunked into transactions by the platform, and the transaction size cannot be controlled.
```
